File: backend/src/main/java/com/snackoverflow/toolgether/domain/postavailability/dto/PostAvailabilityResponse.java
package com.snackoverflow.toolgether.domain.postavailability.dto;

import com.snackoverflow.toolgether.domain.postavailability.entity.PostAvailability;
import lombok.Getter;

import java.time.format.DateTimeFormatter;

@Getter
public class PostAvailabilityResponse {
    private String date;
    private String startTime;
    private String endTime;
    private boolean isRecurring;
    private Integer recurrenceDays;

    public PostAvailabilityResponse(PostAvailability postAvailability, DateTimeFormatter formatter) {
        this.date = postAvailability.getDate() != null ? postAvailability.getDate().format(formatter) : null;
        this.startTime = postAvailability.getStartTime() != null ? postAvailability.getStartTime().format(formatter) : null;
        this.endTime = postAvailability.getEndTime() != null ? postAvailability.getEndTime().format(formatter) : null;
        this.isRecurring = postAvailability.isRecurring();
        this.recurrenceDays = postAvailability.getRecurrence_days();
    }
}

File: backend/src/main/java/com/snackoverflow/toolgether/domain/postavailability/dto/PostAvailabilityResponse.kt
package com.snackoverflow.toolgether.domain.postavailability.dto

import com.snackoverflow.toolgether.domain.postavailability.entity.PostAvailability
import lombok.Getter
import java.time.format.DateTimeFormatter

data class PostAvailabilityResponse(
    val postAvailability: PostAvailability,
    val formatter: DateTimeFormatter
) {
    val date: String? = postAvailability.date?.format(formatter)
    val startTime: String? = postAvailability.startTime?.format(formatter)
    val endTime: String? = postAvailability.endTime?.format(formatter)
    val isRecurring: Boolean = postAvailability.isRecurring
    val recurrenceDays: Int = postAvailability.recurrence_days
}
